### DISCUSSION TITLE
How Do We Write Automated Test Cases for pixi or canvas based apps?

### DISCUSSION THREAD

**Discussion by Amansaxena001** [pixijs/pixijs#10788] [Ideas]
Hi Guys, I need your opinions how do products like figma, excalidraw test their application using automation testing frameworks like cypress, playwright and all.

The main issue where I my mind is going nuts is how do I determine at which position which elements are present. The app has a lot of elements and different actions can be performed on the elements like click, drag etc. Some of the actions are very specific to elements so its very necessary we determine the element and then trigger the interactions. 

One brute-force solution is to dispatch the coordinates from the app but that is not scalable especially when the app has around 1-2 hundred thousands elements.

Please add your thoughts in the comments it will be really helpful

**Comment by Amansaxena001** [comment]
@bigtimebuddy can you pls share any thoughts around this?
@ShukantPal

**Reply by bigtimebuddy** [reply]
Sorry, I cannot respond to every request. If you haven't already, I would suggest you join the [Discord](https://discord.com/invite/QrnxmQUPGV) as that is also a good place to get advice/suggestions.
